frappe/core/doctype/user/user.py

```diff
@@ -279,12 +283,16 @@ def reset_password(self, send_email=False, password_expired=False):
 		return link
 
 	def get_other_system_managers(self):
-		return frappe.db.sql("""select distinct `user`.`name` from `tabHas Role` as `user_role`, `tabUser` as `user`
-			where user_role.role='System Manager'
-				and `user`.docstatus<2
-				and `user`.enabled=1
-				and `user_role`.parent = `user`.name
-			and `user_role`.parent not in ('Administrator', %s) limit 1""", (self.name,))
+		user_doctype = DocType("user").as_("user")
+		has_role_doctype = DocType("Has Role").as_("user_role")
+		return frappe.qb.from_(user_doctype
+			).from_(has_role_doctype
+			).select(user_doctype.name
+			).where(user_doctype.docstatus<2
+			).where(user_doctype.enabled==1
+			).where(has_role_doctype.parent == user_doctype.name
+			).where(has_role_doctype.parent.notin(["Administrator", self.name])
+			).limit(1).distinct()
```

Comment:
```suggestion
		return frappe.qb.from_(user_doctype)
			.from_(has_role_doctype)
			.select(user_doctype.name)
			.where(user_doctype.docstatus<2)
			.where(user_doctype.enabled==1)
			.where(has_role_doctype.parent == user_doctype.name)
			.where(has_role_doctype.parent.notin(["Administrator", self.name]))
			.limit(1)
			.distinct()
```

I personally don't like starting a new line with a bracket... the suggestion looks cleaner to me.

@gavindsouza, @leela is there any convention followed in python for such a case?

Reply:
So I wrote it like that to avoid enclosing the whole thing in brackets.

Like this
```py
		return (frappe.qb.from_(user_doctype)
			.from_(has_role_doctype)
			.select(user_doctype.name)
			.where(user_doctype.docstatus < 2)
			.where(user_doctype.enabled == 1)
			.where(has_role_doctype.parent == user_doctype.name)
			.where(has_role_doctype.parent.notin(["Administrator", self.name]))
			.limit(1)
			.distinct()
			)
```

 Python won't let you chain them otherwise.